gstreamer-rtsp-server/src/r_t_s_p_session_pool.rs

```diff
@@ -0,0 +1,77 @@
+use std::cell::RefCell;
+use std::mem::transmute;
+use RTSPSessionPool;
+use ffi;
+use glib;
+use glib_ffi;
+use glib::object::IsA;
+use glib::translate::*;
+use glib::source::{CallbackGuard, Continue, Priority};
+use glib_ffi::{gboolean, gpointer};
+
+unsafe extern "C" fn trampoline_watch(
+    pool: *mut ffi::GstRTSPSessionPool,
+    func: gpointer,
+) -> gboolean {
+    let _guard = CallbackGuard::new();
+    #[cfg_attr(feature = "cargo-clippy", allow(transmute_ptr_to_ref))]
+    let func: &RefCell<Box<FnMut(&RTSPSessionPool) -> Continue + Send + 'static>> = transmute(func);
+    (&mut *func.borrow_mut())(&from_glib_borrow(pool)).to_glib()
+}
+
+unsafe extern "C" fn destroy_closure_watch(ptr: gpointer) {
+    let _guard = CallbackGuard::new();
+    Box::<RefCell<Box<FnMut(&RTSPSessionPool) -> Continue + Send + 'static>>>::from_raw(
+        ptr as *mut _,
+    );
+}
+
+fn into_raw_watch<F: FnMut(&RTSPSessionPool) -> Continue + Send + 'static>(func: F) -> gpointer {
+    #[cfg_attr(feature = "cargo-clippy", allow(type_complexity))]
+    let func: Box<RefCell<Box<FnMut(&RTSPSessionPool) -> Continue + Send + 'static>>> =
+        Box::new(RefCell::new(Box::new(func)));
+    Box::into_raw(func) as gpointer
+}
+
+pub trait RTSPSessionPoolExtManual {
+    fn create_watch<'a, N: Into<Option<&'a str>>, F>(
+        &self,
+        name: N,
+        priority: Priority,
+        func: F,
+    ) -> Option<glib::Source>
+    where
+        F: FnMut(&RTSPSessionPool) -> Continue + Send + 'static;
+}
+
+impl<O: IsA<RTSPSessionPool>> RTSPSessionPoolExtManual for O {
+    fn create_watch<'a, N: Into<Option<&'a str>>, F>(
```

Comment:
Why did you create this one but not the one for GstRTSPServer?

Reply:
`gst_rtsp_server_create_source`? If so, because it's not generated by gir either